update-assets-docker.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+set -ex
+
+# Default values
+IMAGE_NAME="simulator"
+REUSE_IMAGE=false
+
+# Parse command line arguments
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    --reuse-image)
+      REUSE_IMAGE=true
+      ;;
+    *)
+      echo "Invalid argument: $1"
+      exit 1
+      ;;
+  esac
+  shift
+done
+
+# Check if the Docker image exists locally
+if ! "$REUSE_IMAGE" || ! docker image inspect "$IMAGE_NAME" &> /dev/null; then
+  echo "Docker image $IMAGE_NAME does not exist or --reuse-image flag not provided. Building..."
+  docker build -t $IMAGE_NAME -f simulator/Dockerfile .
+else
+  echo "Docker image $IMAGE_NAME exists, and --reuse-image flag is set. Skipping build."
+fi
+
+for crate_name in "faucet" "radiswap" "flash_loan" "genesis_helper" "metadata" "test_environment" "global_n_owned" "kv_store"
+do
+  echo "Building $crate_name..."
+  docker run -it --entrypoint=scrypto -v $PWD:/src simulator build --path assets/blueprints/$crate_name
```

Comment:
I've no idea if this is possible, but can we make it so that we can provide parameters to the image instead of needing to override the endpoint?

Or maybe we map $PWD/assets/blueprints/$crate_name:/src and it works without overriding the entrypoint?

Reply:
oh, the idea is that people won't need to override the entrypoint. they should just run 
`docker run -v $PWD:/src simulator`
the problem here is that assets have local dependencies that use relative paths so I can't just mount the asset to be built, I need the dependent dirs as well, that's why I ended up bind mounting the $PWD and overriding the entrypoint

Reply:
will add a README with the steps people should run